### ISSUE TITLE
Pretrained weights issue  - what is proper usage

### ISSUE THREAD

**Issue by princealy** [Orion-AI-Lab/EfficientBigEarthNet#2]
Hey! 

Amazing work with the all the models created in this repo! I am trying to load the model and apply the weights. I believe this is similar to the other issue reported but I am trying to use one of the custom classes created in the repo. Here is my code

```
import tensorflow as tf
from tensorflow import keras
import sys
sys.path.append('/content/')  # Replace with your directory path
import models


model_name = "EfficientNetB0"
model = getattr(models, model_name)(
            nb_class=43, modality="MM").model
print(model.summary())
checkpoint_dir = "folder_location/checkpoint_EfficientNetB0/"
latest = tf.train.latest_checkpoint(checkpoint_dir)
model.load_weights(latest)
```

I got the weights for `checkpoint_EfficientNetB0` from the dropbox. 

I get the follow issue:
`AssertionError: Nothing except the root object matched a checkpointed value. Typically this means that the checkpoint does not match the Python program. The following objects have no matching checkpointed value: [<tf.Variable 'top_conv/kernel:0' shape=(1, 1, 320, 1280)`

I am working on loading the weights, so I can save as an `ONNX` model and so it can be used with Pytorch as well. Thanks for your help in advance.

**Comment by princealy** [comment]
I have also tried the following with the same error
```
from tensorflow.keras.applications import VGG16, VGG19, ResNet50, ResNet101, ResNet152


model = ResNet50(
            include_top=True,
            weights=None,
            input_shape=(120, 120, 10),
            pooling="avg",
            classes=19
)
checkpoint_dir = "folder_location/checkpoint_ResNet50/"
latest = tf.train.latest_checkpoint(checkpoint_dir)
model.load_weights(latest)

```

**Comment by azavras** [comment]
Hi @princealy,
i kindly ask you to read again my response to the issue #1. All in all, on your first comment you should correct the nb_class and modality parameters, while on your second comment you should use our ResNet50 model instead of the Keras one.

**Comment by princealy** [comment]
Hey @azavras !

Thanks for getting back to me. I tried following your response in the issue but unfortunately, I am still facing the same problem. 


For `EfficientNetB0`
I run the following:
```
model = models.EfficientNetB0(12, "MM").model
checkpoint_dir = "folderPath/checkpoint_EfficientNetB0"
latest = tf.train.latest_checkpoint(checkpoint_dir)
model.load_weights(latest)
```
And I tried different variations, such as nb_class = 10 and Modality parameter =  S2

Then with the resnet, I also tried the following:

```
model = models.ResNet50BigEarthModel(10, "S2").model
checkpoint_dir = "/folderPAth/checkpoint_ResNet50/"
latest = tf.train.latest_checkpoint(checkpoint_dir)
model.load_weights(latest)
```
And I downloaded the model from this [link](https://www.dropbox.com/s/trsojsbjvlpt9jz/checkpoint_resnet50.zip?dl=0)

Maybe I am using the model incorrectly or the wrong modality parameter, but not sure where I should find instructions for that. Thanks a lot for your help.
